frontend/src/components/OtherImplementations.tsx

```diff
@@ -74,3 +99,12 @@ const getImplementationPath = (id: string): string => {
   const pathSegment = id.split("/");
   return pathSegment[pathSegment.length - 1];
 };
+
+const getLatestSupportedDialect = (impl: Implementation): string => {
+  const latestSupportedDialect = impl.dialects.sort((a, b) =>
+    b.localeCompare(a)
```

Comment:
You sort it by name which not necessarily might be a correct way for finding the latest dialect. You can first map the `dialects` array elements to instances of Dialect (`Dialect.foPath()`) and then sort it. Or even better, because you actually don't need to sort the whole array, just find min element (by publish date). `reduce()` can do that stuff

Reply:
Actually, `Dialect.foPath()` won't work, so bigger changes are required :) Why not change the type of `Implementation.dialects` to be `Dialect[]` and assign it when parsing implementation data?

In `Dialect.ts` you can provide an additional method `forUri()` - don't really see a very clean way of implementing this, but you can figure it out

Reply:
Which approach do you suggest I take here ? I have already added `forURI()`  as you pointed out below. I can make the changes you suggested below but overall what are you inclined towards ?

Reply:
We could use `find()` as well right ?